[2025-10-06T14:52:35.750+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-06T14:52:35.764+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: main.run_pyspark_task manual__2025-10-06T14:52:34.529276+00:00 [queued]>
[2025-10-06T14:52:35.769+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: main.run_pyspark_task manual__2025-10-06T14:52:34.529276+00:00 [queued]>
[2025-10-06T14:52:35.770+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-10-06T14:52:35.780+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): run_pyspark_task> on 2025-10-06 14:52:34.529276+00:00
[2025-10-06T14:52:35.785+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=9097) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-10-06T14:52:35.787+0000] {standard_task_runner.py:63} INFO - Started process 9099 to run task
[2025-10-06T14:52:35.787+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'main', 'run_pyspark_task', 'manual__2025-10-06T14:52:34.529276+00:00', '--job-id', '156', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmp7eu88vel']
[2025-10-06T14:52:35.789+0000] {standard_task_runner.py:91} INFO - Job 156: Subtask run_pyspark_task
[2025-10-06T14:52:35.820+0000] {task_command.py:426} INFO - Running <TaskInstance: main.run_pyspark_task manual__2025-10-06T14:52:34.529276+00:00 [running]> on host 5adecdfd8c83
[2025-10-06T14:52:35.879+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='main' AIRFLOW_CTX_TASK_ID='run_pyspark_task' AIRFLOW_CTX_EXECUTION_DATE='2025-10-06T14:52:34.529276+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-06T14:52:34.529276+00:00'
[2025-10-06T14:52:35.880+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-06T14:52:38.289+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-06T14:52:38.290+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main.py", line 24, in run_spark_job
    .master("local") \
     ^^^^^^
AttributeError: 'SparkSession' object has no attribute 'master'
[2025-10-06T14:52:38.298+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=main, task_id=run_pyspark_task, run_id=manual__2025-10-06T14:52:34.529276+00:00, execution_date=20251006T145234, start_date=20251006T145235, end_date=20251006T145238
[2025-10-06T14:52:38.309+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 156 for task run_pyspark_task ('SparkSession' object has no attribute 'master'; 9099)
[2025-10-06T14:52:38.330+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-10-06T14:52:38.340+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-06T14:52:38.343+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
